monai/transforms/compose.py

```diff
@@ -107,7 +107,11 @@ def execute_compose(
 
     for _transform in transforms[start:end]:
         if threading:
-            _transform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
+            if isinstance(_transform, ThreadUnsafe):
+                if isinstance(_transform, Randomizable):
+                    # update the random state before deepcopy, otherwise there is no randomness
+                    _transform.randomize(data)
```

Comment:
We can definitely update the random state here, but I guess the issue here is that if the transform is thread unsafe, we can't guarantee that the same transform will be performed on all keys, which may cause problems.

Reply:
As of my understanding, the state is frozen for a single thread after the subsequent deepcopy of the Transform. Since all keys are processed by this copied Transform, a consistent state is guaranteed.

Reply:
Actually, I realized that `.randomize()` is not necessarily updating the random state `self.R` (cf. `monai.transforms.transform.RandomizableTranform`)
![image](https://github.com/user-attachments/assets/e53d2a8d-bad3-46e5-b99c-81e0fb05d278)

Therefore the correct way here would be to call the `_transform.set_random_state()` which is implemented in the  `Randomizable` base class und updates `self.R`

Reply:
@KumoLiu are there more transforms which inherit directly from `ThreadUnsafe`? I can only find `Randomizable` in the monai codebase, which would be covered here.